addons/mail/static/src/core/web/store_service_patch.js

```diff
@@ -126,5 +126,26 @@ const StorePatch = {
         this.store.starred.messages = [];
         await this.env.services.orm.call("mail.message", "unstar_all");
     },
+    handleClickOnLink(ev, thread) {
+        const model = ev.target.dataset.oeModel;
+        const id = Number(ev.target.dataset.oeId);
+        if (
+            !["o_mail_redirect", "o_channel_redirect"].includes(ev.target.classList.value) &&
+            ev.target.tagName === "A" &&
+            id
+        ) {
+            ev.preventDefault();
+            Promise.resolve(
+                this.env.services.action.doAction({
+                    type: "ir.actions.act_window",
+                    res_model: model,
+                    views: [[false, "form"]],
+                    res_id: id,
+                })
+            ).then(() => this.onLinkFollowed(thread));
+            return true;
+        }
+        return super.handleClickOnLink(ev, thread);
+    },
```

Comment:
It's not ok to copy-paste the conditional of original code to put an exclusive branching here: if the conditional changes in original code, this part is likely forgotten and it's painful to maintain. We should avoid repeating ourselves, and conditional should focus on its concern and not others.

The `action.doAction()` is a fallback, so should be called after super returns `false` (meaning this was not handled in original code).